### ISSUE TITLE
Image for 1.24 is removed

### ISSUE THREAD

**Issue by davesmits** [tinglesoftware/dependabot-azure-devops#1206]
Seems 1.24 is removed. Any chance to get it back?

error message:

```
Unable to find image 'ghcr.io/tinglesoftware/dependabot-updater-nuget:1.24' locally
docker: Error response from daemon: manifest unknown.
```

**Comment by rhyskoedijk** [comment]
I get the feeling a lot of people still use 1.24 as it was the last release before they changed the [NuGet] updater to use the native helpers, which drastically increased the run times.

@mburumaxwell do you know why this got removed?
Looking at `cleanup.yml`, I see you've set ignore version rules that should have stopped "1.24" from deleting, but it seems this didn't work?
```yml
# exclude major and minor versions as per SemVer e.g. 1.20
ignore-versions: '^([0-9]\\d*)\\.([0-9]\\d*)$'
```

**Comment by davesmits** [comment]
@rhyskoedijk beside the run times never got authentication to work neither.
also we hacked a kind of grouping by using a variable for nuget versions where the version was the same; which also was broken. 

So tbh; github did a horrible job if you ask me :). looking forward for your enhancements!

**Comment by rhyskoedijk** [comment]
> it's unfortunate for auth. In as much as they say, ADO will continue to exist alongside GitHub, the actions may be speaking contrary.

This is a little off topic sorry but it reminded me that the [Azure DevOps Roadmap](https://learn.microsoft.com/en-us/azure/devops/release-notes/features-timeline#github-advanced-security-for-azure-devops) mentions [Automatically fix detected dependency scanning vulnerabilities with Dependabot security updates](https://learn.microsoft.com/en-us/azure/devops/release-notes/roadmap/2024/ghazdo/dependabot) due 2024 Q2. This makes me think that Microsoft _must_ be working on getting Dependabot working natively in DevOps so that they can offer it to users who pay for Advanced Security in DevOps (extra $49 per active committer per month). 

Hopefully that means ADO authentication issue gets resolved too; unless they're going to roll it out without support for private package feeds. I just hope Microsoft end up patching the issues in dependabot-core, rather than implementing the fix in their own infrastructure.

If Microsoft does paywall Dependabot behind Advanced Security for DevOps, this project is going to have a long life yet as why pay for it when you can get a free marketplace extension right? :stuck_out_tongue_winking_eye:

**Comment by Lukejkw** [comment]
Any idea on timelines for address this issue @mburumaxwell ?

We rely on version 1.24 because of the authentication issues.

**Comment by mburumaxwell** [comment]
No

**Comment by mburumaxwell** [comment]
Is there still a need for the 1.24 image? Major contributions by @rhyskoedijk have been released in [1.30.0](https://github.com/tinglesoftware/dependabot-azure-devops/releases/tag/1.30.0) plus the workaround provided in https://github.com/tinglesoftware/dependabot-azure-devops/issues/921#issuecomment-2162273558

**Comment by ryangribble** [comment]
Hiya 👋   

I'm on the same team as @Lukejkw and just chiming in to say that the new 1.30 version did still suffer from the auth issue on Azure DevOps private NuGet feeds (which was the reason we were pinned back on 1.24 in the first place)... but the mentioned workaround to install the NuGet credential manager and pass the pipeline's AccessToken to it worked!

So yes 1.30 + the workaround looks like it's viable 🥳 

Very excited to try out the Grouping feature.  Thanks heaps @rhyskoedijk  and @mburumaxwell 🙌

**Comment by andrewimcclement** [comment]
Hmm, I'm hitting a different issue on 1.30 (doesn't find Directory.Packages.props and try to update it). I'll aim to post an issue with a reproduction sometime next week. Unfortunately, until I can work out a fix/workaround, I would still want 1.24, since that is known to work. (I would love to upgrade to 1.30 if I could.)

**Comment by andrewimcclement** [comment]
Hmm, my issue only seems to occur when in repositories using central package management but with all sln files deeper in the directory tree. If the sln file is found at the top level, the workaround is fine. So honestly I'm sorted for 90% of my repositories.

**Comment by rhyskoedijk** [comment]
> doesn't find Directory.Packages.props and try to update it

A quick scan of the [dependabot-core issues list](https://github.com/dependabot/dependabot-core/issues?q=is%3Aissue+is%3Aopen+Directory.Packages.props) show there are a few open mentioning `.prop` files maybe one of them is related to your issue? I don't have much experience with them personally but if you've got logs and config files to share I can have a quick look in to it.

> my issue only seems to occur when in repositories using central package management but with all sln files deeper in the directory tree. If the sln file is found at the top level, the workaround is fine. 

I might not understand your issue, but now that there is [support for multiple directories](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directories), you might be able to update your `dependabot.yml` to define all directories that contain `.sln` files if you are finding Dependabot is discovering them otherwise.

If anybody else is testing out group PR support in 1.30, I have just  found a bug that will cause group PRs to get closed on subsequent runs of the update script. The fix is in https://github.com/tinglesoftware/dependabot-azure-devops/pull/1222 but you can temporarily work around this by disabling the "Abandon unwanted pull requests" until the fix is released.

**Comment by brechtvlaanderen** [comment]
> Hiya 👋
> 
> I'm on the same team as @Lukejkw and just chiming in to say that the new 1.30 version did still suffer from the auth issue on Azure DevOps private NuGet feeds (which was the reason we were pinned back on 1.24 in the first place)... but the mentioned workaround to install the NuGet credential manager and pass the pipeline's AccessToken to it worked!
> 
> So yes 1.30 + the workaround looks like it's viable 🥳
> 
> Very excited to try out the Grouping feature. Thanks heaps @rhyskoedijk and @mburumaxwell 🙌

Can someone share a full end to end example of the workaround? I'm not succeeding in migrating to from 1.24 to 1.30 using the workaround. I tried all kinds of combinations. I had this before as depandabot.yml with a devops variable PAT. With the workaround I keep getting: _The following source could not be reached as it requires authentication (and any provided details were invalid or lacked the required permissions)._Should I keep setting the token in the depenabot.yml file also? I already tried using an environment variable with the system accestoken,.. Can anybody share a full working example?

`version: 2
registries:
  artifacts-dbs:
    type: nuget-feed
    url: https://pkgs.dev.azure.com/xxx/xxx/_packaging/xxx/nuget/v3/index.json
    token: PAT:${{xxx-artifacts-pat}}
updates:
- package-ecosystem: "nuget"
  directory: "/src"
  open-pull-requests-limit: 15
  target-branch: "master"
  registries:
    - xxx-xxx
  groups: #not yet working https://github.com/tinglesoftware/dependabot-azure-devops/issues/685
    dependencies:
      patterns:
      - "Microsoft.AspNetCore*"
      - "Microsoft.EntityFrameworkCore*"
      - "Microsoft.Extensions.*"
      - "Microsoft.ServiceFabric.*"`

**Comment by rhyskoedijk** [comment]
> Can someone share a full end to end example of the workaround? I'm not succeeding in migrating to from 1.24 to 1.30 using the workaround.

I have just put up a change (https://github.com/tinglesoftware/dependabot-azure-devops/pull/1233) that will make ADO NuGet feed auth "just work" like it did in 1.24, no special workaround required. I think this workaround is a bit complicated and ideally it could just work by using the auth token you've defined in `dependabot.yml`.

**Comment by rhyskoedijk** [comment]
> I have just put up a change (https://github.com/tinglesoftware/dependabot-azure-devops/pull/1233) that will make ADO NuGet feed auth "just work" like it did in 1.24

This is now released in 1.30.2; anybody that pinned 1.24 due to NuGet auth issues should give 1.30.2 a try.

**Comment by brechtvlaanderen** [comment]
> > I have just put up a change (#1233) that will make ADO NuGet feed auth "just work" like it did in 1.24
> 
> This is now released in 1.30.2; anybody that pinned 1.24 due to NuGet auth issues should give 1.30.2 a try.

Yes it works. Big thank you!

**Comment by mburumaxwell** [comment]
The amount of effort required to restore 1.24 is higher than necessary. Seeing that both scripts have been updated, I'll proceed to close this issue.
